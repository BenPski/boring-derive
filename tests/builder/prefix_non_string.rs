use boring_derive::Builder;

#[derive(Debug, Default, Builder)]
#[builder(prefix = 1)]
struct Example {
    item: String,
}

fn main() {
    let ex = Example::default().set_item("balls");
    println!("{:?}", ex);
}
