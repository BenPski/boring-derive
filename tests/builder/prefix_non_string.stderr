error: prefix must be a string not `1`
 --> tests/builder/prefix_non_string.rs:4:11
  |
4 | #[builder(prefix = 1)]
  |           ^^^^^^^^^^

error[E0599]: no method named `set_item` found for struct `Example` in the current scope
  --> tests/builder/prefix_non_string.rs:10:33
   |
5  | struct Example {
   | -------------- method `set_item` not found for this struct
...
10 |     let ex = Example::default().set_item("balls");
   |                                 ^^^^^^^^ method not found in `Example`
